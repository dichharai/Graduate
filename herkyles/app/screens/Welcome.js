import React, { Component } from "react"; 
import { View, Text, StyleSheet, Button, AsyncStorage, StatusBar} from "react-native"; 

import {Container} from "../components/Container"; 
import {firebaseApp} from "../../db/DbConfig"; 
import {LinkBtns} from "../components/Buttons/LinkBtns"; 
import {LinkTouch} from '../components/Buttons/LinkTouch';

class Welcome extends Component{
    static navigationOptions = {
        title: "Welcome", 
        headerStyle: {
            backgroundColor: "#000000",
        }, 
          headerTintColor: '#facf33',
    };

    constructor(props){
        super(props);
        this.state = { userEmail: ""};
        this._getUserEmail();
    }
    handleRecOptionPress = () => {
        this.props.navigation.navigate("RecOptions"); 
        console.log("RecOption Press"); 
    }

    _getUserEmail = async () => {
        try{
            let email = await AsyncStorage.getItem("userEmail");
            console.log("userEmail in WelcomeAdmin: " + email);
            this.setState({userEmail: email});
            console.log("email from state: "+ this.state.userEmail);
            
        }catch(e){
            console.log(e);
        }
    }
    handleLogout = async () => {
        try{
            await firebaseApp.auth().signOut();


            await AsyncStorage.clear(); 
            let userToken = await AsyncStorage.getItem("userToken");
            console.log("userToken after clearing: " + userToken);
            let userEmail = await AsyncStorage.getItem("userEmail");
            console.log("userEmail after clearing: " + userEmail);
            
            this.props.navigation.navigate("App");
        }catch(e){
            console.log(e);
        }
    }

    render(){
        return (
                <Container>
                    <StatusBar transclucent={false} barStyle="light-content"/>
                    <Text>Welcome Admin</Text>
                
                    <Text> Hello {this.state.userEmail}</Text>
                    {console.log("userEmail in render: " + this.state.userEmail)}
                    <LinkBtns
                    text="Manage Equipments"
                    onPress={this.handleRecOptionPress}
                    />
                    <LinkTouch 
                    text="Logout"
                    onPress={this.handleLogout} />
                </Container>            
        );
    }
}

export default Welcome; 