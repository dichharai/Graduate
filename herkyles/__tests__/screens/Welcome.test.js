import React from "react";
import Welcome from "../../app/screens/Welcome";
import MockAsyncStorage from 'mock-async-storage';

import renderer from "react-test-renderer";
import { storage } from "firebase";
import {AsyncStorage as aStorage } from 'react-native'; 

const mock = () => {
  const mockImpl = new MockAsyncStorage(); 
  jest.mock('AsyncStorage', () => mockImpl);
}
mock(); 


describe ("Welcome", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<Welcome />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
  
  it('Mock Async Storage working', async () => {
    await aStorage.setItem('userEmail', 'abc@gmail.com'); 
    const value = await aStorage.getItem('userEmail');
    expect(value).toBe('abc@gmail.com');
  });
  
  it('should call Logout', async () => {
    //let handleLoginPressMock = jest.fn(); 
    //let navigationMock = {navigate: jest.fn()}
    const navigationMock = {navigate: jest.fn()}; 

    let welcomeComponent = renderer.create(<Welcome 
        navigation={navigationMock}
      />
    ).getInstance();
  
    welcomeComponent.handleLogout(); 
    expect(navigationMock.navigate).toHaveBeenCalledTimes(0); 
     

  });
  
  /*
  it('it should call _getUserEmail()', async () => {
    const navigationMock = {navigate: jest.fn()}; 

    let authComponent = renderer.create(<AuthLoading 
        navigation={navigationMock}
      />
    ).getInstance(); 

    authComponent._bootstrapAsync(); 
    await expect(navigationMock.navigate).toHaveBeenCalledTimes(0); 
 
  }); 
  */
});