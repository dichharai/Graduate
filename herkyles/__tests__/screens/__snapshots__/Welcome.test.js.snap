// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Welcome Admin
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Hello 
    
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Manage Equipments
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
